src/modules/Elsa.EntityFrameworkCore/Modules/Management/WorkflowDefinitionStore.cs

```diff
@@ -60,6 +60,12 @@ public async Task<IEnumerable<WorkflowDefinition>> FindManyByDefinitionIdAsync(s
         return await _store.FindAsync(predicate, Load, cancellationToken);
     }
 
+    public async Task<IEnumerable<WorkflowDefinition>> FindByPredicateAsync(Expression<Func<WorkflowDefinition, bool>> predicate, VersionOptions versionOptions, CancellationToken cancellationToken = default)
```

Comment:
What if the implementation of `IWorkflowDefinitionStore` does not support translating arbitrary expressions to the underlying storage mechanism? For example, some users may implement an HTTP API client underneath, which makes translating these expressions considerably hard, if not impossible.

Reply:
Good point, I will hide that implementation at a lower level.